src/pulselistener/pulselistener/mercurial.py

```diff
@@ -228,6 +229,6 @@ def clean(self):
         logger.info('Diff has been pushed !')
 
         # Publish Treeherder link
-        if build_target_phid and self.publish_treeherder_link:
+        if build_target_phid and self.publish_phabricator:
```

Comment:
Since it's working well, I think we can remove the optionto publish the Treeherder link and only use it for the failure publication.
After that works in prod too for a while, we can completely remove the option

Reply:
This option is needed to make a shadow mode version of the bot, to run on testing.

I was thinking of simplifying this by using a `read-only` or `read-write` mode in our `PhabricatorApi` client.
Methods that post data on Phabricator would check if the client configuration and only proceed if it's in `read-write`.
That would be another PR ...